### PR TITLE
[FEATURE] @pipedream/platform axios should support the ability to create custom axios constructor, or pass all standard axios options #5497

### PR DESCRIPTION
**Pull request by lcaresia**

### CODE REVIEW

**Comment by vunguyenhung** [comment]
Hello @lcaresia @andrewjschuang, would you mind advice how can I test this? Or is this needed testing?

**Comment by andrewjschuang** [comment]
@vunguyenhung I think you can create two instances of each `axios`, e.g.
```js
import axios from "axios";
import { axios as pipedreamAxios } from "./platform/dist/index.js";
```
or something like that.

And then create a config to pass in to the constructor with most or all `axios` parameters, make the request and see if they were really used by `pipedreamAxios`.

**Comment by vunguyenhung** [comment]
Hello @andrewjschuang, I couldn't test the action due to an error on the publish action step. 

Could you check the test report below and advice how can I test it?
[Axios_5544_1037.pdf](https://github.com/PipedreamHQ/pipedream/files/10835843/Axios_5544_1037.pdf)

**Comment by andrewjschuang** [comment]
Hi @lcaresia, looking at the code, seems like the change doesn't reflect in anything in practice, since we don't remove any key from the `config` object.

Also, looking at [the thread](https://pipedream.com/community/t/can-i-use-axios-directly-with-pipedream/4774/2) that originated this issue, I think it's if best we implemente the [axios.create()](https://axios-http.com/docs/instance) method.

I also thought of adding a flag `fullResponse` with `default: false` in `AxiosRequestConfig` that enables the user to receive the whole [response schema](https://axios-http.com/docs/res_schema) from the request instead of just `data`, what do you think?

**Comment by lcaresia** [comment]
@vunguyenhung @andrewjschuang 

You can test creating an `test.js` file on project root folder with this code:

```
const {
  axios, createAxiosInstance,
} = require("./platform/dist");

async function test1() {
  const createdAxios = createAxiosInstance({
    baseURL: "https://random-data-api.com/api/v2",
  });

  const response = await createdAxios.get("users");

  console.log("createdAxios:", response);
}

async function test2() {
  const data = await axios(this, {
    url: "https://random-data-api.com/api/v2/users",
    returnRawResponse: false,
  });

  console.log("onlyData:", data);
}

async function test3() {
  // const axios = createAxiosInstance({});

  const response = await axios(this, {
    url: "https://random-data-api.com/api/v2/users",
    returnRawResponse: true,
  });

  console.log("returnRawResponse:", response);
}

test1();
test2();
test3();
```

File: platform/lib/index.ts [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import * as t from "io-ts";
 
-import axios from "./axios";
+import axios, { createAxiosInstance } from "./axios";
 import { AxiosRequestConfig as AxiosConfig } from "axios";
 
 export {
   axios,
+  createAxiosInstance,
```
Comment (andrewjschuang) [inline platform/lib/index.ts:8]:
```suggestion
```
See other comment about `axios.create`

File: platform/lib/index.ts [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import * as t from "io-ts";
 
-import axios from "./axios";
+import axios, { createAxiosInstance } from "./axios";
```
Comment (andrewjschuang) [inline platform/lib/index.ts:3]:
```suggestion
```
See other comment about `axios.create`

**Review by andrewjschuang — CHANGES_REQUESTED** [review]
Thanks @lcaresia!
I've left some comments about some improvements, but overall this is in the right track!

File: platform/lib/axios.ts [hunk 3]
```diff
@@ -5,6 +5,10 @@ import * as querystring from "querystring";
 import { cloneSafe } from "./utils";
 import { ConfigurationError } from "./errors";
 
+interface PipedreamAxiosRequestConfig extends AxiosRequestConfig {
```
[the remark this answers is not in the document]
Comment (lcaresia) [reply platform/lib/axios.ts:8]: Oh, I didn't see that

**Review by andrewjschuang — APPROVED** [review]

**Pull request merged**